packages/twenty-server/src/engine/core-modules/email-verification/services/email-verification.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { render } from '@react-email/render';
+import { addMilliseconds, differenceInMilliseconds } from 'date-fns';
+import ms from 'ms';
+import { SendEmailVerificationLinkEmail } from 'twenty-emails';
+import { Repository } from 'typeorm';
+
+import {
+  AppToken,
+  AppTokenType,
+} from 'src/engine/core-modules/app-token/app-token.entity';
+import { EmailVerificationTokenService } from 'src/engine/core-modules/auth/token/services/email-verification-token.service';
+import { DomainManagerService } from 'src/engine/core-modules/domain-manager/service/domain-manager.service';
+import {
+  EmailVerificationException,
+  EmailVerificationExceptionCode,
+} from 'src/engine/core-modules/email-verification/email-verification.exception';
+import { EmailService } from 'src/engine/core-modules/email/email.service';
+import { EnvironmentService } from 'src/engine/core-modules/environment/environment.service';
+import { UserWorkspaceService } from 'src/engine/core-modules/user-workspace/user-workspace.service';
+import { UserService } from 'src/engine/core-modules/user/services/user.service';
+
+@Injectable()
+// eslint-disable-next-line @nx/workspace-inject-workspace-repository
+export class EmailVerificationService {
+  constructor(
+    @InjectRepository(AppToken, 'core')
+    private readonly appTokenRepository: Repository<AppToken>,
+    private readonly domainManagerService: DomainManagerService,
+    private readonly emailService: EmailService,
+    private readonly environmentService: EnvironmentService,
+    private readonly userService: UserService,
+    private readonly userWorkspaceService: UserWorkspaceService,
+    private readonly emailVerificationTokenService: EmailVerificationTokenService,
+  ) {}
+
+  async sendVerificationEmail(
+    userId: string,
+    email: string,
+    workspaceSubdomain?: string,
+  ) {
+    if (!this.environmentService.get('IS_EMAIL_VERIFICATION_REQUIRED')) {
+      throw new EmailVerificationException(
+        'Email verification token cannot be sent because email verification is not required',
+        EmailVerificationExceptionCode.EMAIL_VERIFICATION_NOT_REQUIRED,
```

Comment:
Why handle this as an error? If email verification isn't require it should just pass silently no?

Reply:
The frontend uses the same env to conditionally invoke the mutation that leads to `sendVerificationEmail()`. This made sense to me as it would avoid unnecessary calls to the backend when `IS_EMAIL_VERIFCATION_REQUIRED` is `false`.

To answer your question, I am throwing this here and in some other methods because I thought it might confuse the users if, for instance, they're trying to verify their email or resend the email verification token but it seems like it's not working (cause it's passing silently) and they don't realize the env has been reset or not set to true in the first place.

Reply:
I see! Thanks

> The frontend uses the same env to conditionally invoke the mutation that leads to sendVerificationEmail().

This is not something that you would typically do on the frontend. If you have doubts, usually the more goes in the backend the better! (because tomorrow their might be a mobile app or other frontends).

Make sure that this error isn't triggered for someone who have `IS_EMAIL_VERIFCATION_REQUIRED` set to `false`. I haven't pulled the code yet (will do), but when @Bonapara did he saw this exception (with IS_EMAIL_VERIFCATION_REQUIRED not set initially)

Reply:
Resolving this since it is now handled silently.